pkg/controller/job/job_controller.go

```diff
@@ -1696,6 +1696,10 @@ func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType
 func recordJobPodFinished(job *batch.Job, oldCounters batch.JobStatus) {
 	completionMode := completionModeStr(job)
 	diff := job.Status.Succeeded - oldCounters.Succeeded
+	// prometheus CounterVec metric must be monotonically increasing
```

Comment:
nit: I would suggest to expand the comment here to say that the counter for succeeded pods may decrease when an indexed job is scaled down.

Reply:
The counter actually won't be decreased when the indexed job is scaled down, since the `diff` would be negative and thus set to 0 on lines 1700-1702. The prometheus counter metric will actually panic if you try to decrease it, which is why I update it by max(0, diff).

Reply:
Let's see this scenario:
1. completions = 6, and newSucceededPods={4}
2. metric=1
3. Then, user changes completions=3 and newSucceededPods={2}

The metric should be 2, not 1, because {2} is a newly succeeded pod. Can you fix that?

Reply:
Ok I figured out a way to address this scenario and updated the code accordingly, let me know what you think.